app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -141,9 +141,12 @@ private void updateHighlighting(final boolean recompute) {
 
                 // Addition of spans which require reflow can shift text on re-application of spans
                 // we compute the resulting shift and scroll the view to compensate in order to make
-                // the experience smooth for the user
-                final int shiftTestLine = layout.getLineForVertical(_hlRect.centerY());
-                final int oldOffset = layout.getLineBaseline(shiftTestLine);
+                // the experience smooth for the user.
+                int shiftTestLine = -1, oldOffset = -1;
+                if (_scrollView != null && _hlRect.height() == _oldHlRect.height()) {
+                    shiftTestLine = layout.getLineForVertical(_hlRect.centerY());
+                    oldOffset = layout.getLineBaseline(shiftTestLine);
+                }
```

Comment:
Don't try to correct scroll position when we are changing view size (i.e. keyboard shown)